server/src/main/kotlin/ir/armor/tachidesk/impl/Chapter.kt

```diff
@@ -27,96 +27,105 @@ import org.jetbrains.exposed.sql.update
 
 object Chapter {
     /** get chapter list when showing a manga */
-    suspend fun getChapterList(mangaId: Int, onlineFetch: Boolean): List<ChapterDataClass> {
-        return if (!onlineFetch) {
+    suspend fun getChapterList(mangaId: Int, onlineFetch: Boolean?): List<ChapterDataClass> {
+        return if (onlineFetch == true) {
+            getSourceChapters(mangaId)
+        } else {
             transaction {
                 ChapterTable.select { ChapterTable.manga eq mangaId }.orderBy(ChapterTable.chapterIndex to DESC)
                     .map {
                         ChapterTable.toDataClass(it)
                     }
+            }.ifEmpty {
+                // If it was explicitly set to offline dont grab chapters
```

Comment:
why?

Reply:
If you wanted to be specifically offline you could set it to false, could remove that part though